### ISSUE TITLE
Right arm board 2B2 not found

### ISSUE THREAD

**Issue by kt10aan** [robotology/icub-support#204]
##### Description of the failure
During startup, board 2B2 (boardId 3) of the right arm cannot be found. 

##### Detailed conditions when the failure occurred
The arm was working fine, until it was grabbed quite hard from the plastic part. Next startup the problem appeared. Likely reason seems to be a cable disconnection somewhere. In January we had made some repairs to the right arm, and putting the covers back, it is possible that the cables didn't end up in their optimal position under the covers. 

Which cables should we test for continuity?

**Comment by julijenv** [comment]
Hi @kt10aan ,
please follow this procedure in order to understand what is going on:
screenshot of net 2 and 6 and 8 with canLoader 
open the upper arm cover -remove the cover nad skin part
check the behaviour of the lights for 2B2 compare to 2B3, 2B4 when you turn on the motor switch.
take some pictures of the wire at the level of the power supply (rainbow colors connector).
cant you see any wire disconnected?
Let us know

**Comment by kt10aan** [comment]
Here are screenshots of nets 2, 6 and 8
![canloadernet2](https://cloud.githubusercontent.com/assets/1415279/13840314/1e816238-ec65-11e5-96e9-119521c1033d.png)
![canloadernet6](https://cloud.githubusercontent.com/assets/1415279/13840316/22114d96-ec65-11e5-9b97-9c6cdf72d39c.png)
![canloadernet8](https://cloud.githubusercontent.com/assets/1415279/13840318/24f4f85a-ec65-11e5-835c-4512f4ed4c78.png)

Here is a photo of the rainbow connector

![img_0426](https://cloud.githubusercontent.com/assets/1415279/13840386/9deb372e-ec65-11e5-9f7c-e4f2ddb20a4c.JPG)

The cables seem fine and I have pushed the connector all the way in just to make sure.

I also inspected all the connectors of the board. I could see anything being disconnected.

hen starting up the motors, the following behaviour is observed:

All three boards have their side-by-side blue and orange LEDs lit up. But afterwards, boards 2B3 and 2B4 have the blue LED constantly on and the orange one blinking at a slow rate. However, 2B2 after the initial startup whre both blue and orange is lit up, only blue remains lit up and the orange does not light up, nor blinking.

Here is an image, taken when the 2B3 and 2B4 orange LED is lit up during blinking.
The bottom board is the 2B2, and the orange LED is not blinking or lighting up.

![img_0425](https://cloud.githubusercontent.com/assets/1415279/13840495/28a3aa9a-ec66-11e5-8637-86f16d4b1a99.JPG)

**Comment by kt10aan** [comment]
A further update:

Starting the motors and hitting the Connect button to Net2 within a couple of seconds in ``canLoader``, the board shows up. Trying to flash the firmware using
``canLoader --canDeviceType cfw2can --canDeviceNum 2 --boardId 3 --firmware 4DC.1.19.out.S``

fails with 

``\Initializing prompt version of canLoader...``
``Selecting canDeviceType cfw2can canDeviceNum 2, boardId 3, firmware 4DC.1.19.out.S``
``Cfw2Can::open() returning error -2``
``yarpdev: ***ERROR*** driver <cfw2can> was found but could not open``
``[ERROR]no boards found ``
``[ERROR]check canbus cable, power supply connection etc.``

Also, with the above procedure, the blue and the orange LEDs on board 2B2 are staying on (while on the other boards are following the usual pattern "both on at startup, then blue on, orange blinking".

**Comment by julijenv** [comment]
Hi @kt10aan ,
Please remove the 4 screws holding the 3 boards stack (the aluminium L shape) and check the connexion on the power side of the board 2B2 - this board is missing from canLoader so by taking a look to the picture the lights are ON on the board so it means that the CAN line "balck filter" has a cable disconnected.
Please check this board. Let me know

**Comment by traversaro** [comment]
@kt10aan can you try to flash the board using the canLoader graphical interface connected to the board in the first couple of seconds from giving power to the board? This solved a similar issue for me in a similar issue, see https://github.com/robotology/icub-support/issues/205#issuecomment-197931051 .

**Comment by julijenv** [comment]
@kt10aan , try to see if you can get connected to the board when clicking connect on canLoader within 5 seconds after having powering up the boards? this will tell you if he flash of the board could have failed in some point.

**Comment by randaz81** [comment]
Cfw2Can::open() returning error -2
means that you tried to use canLoader while another instance of canLoader (or robotInterface) was running!

**Comment by kt10aan** [comment]
@randaz81, doh, you are right, I forgot that the GUI canLoader was running!
@julijenv, I could not see any cable being disconnected. Also, see my answer above; I could see the board when connecting in canLoader within 5 seconds of starting the motors.
I reflashed the board using the GUI canLoader and it worked! @traversaro , thanks for the tip, I had seen #205 and I was going to try it as last resort before "find the bad cable"

So all the boards are showing up nicely now. Very happy that I did not have to  fix cables. However, I am mystified by the reason for this. Before  the issue appeared, everything was working fine and no firmware updates took place during that time.
How can we be sure for the reliability of the firmware update process? Apart from flashing each board 3 times just to make sure! :)
In any case, issue solved.
